cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js

```diff
@@ -105,6 +109,15 @@ export default class DataPrepTable extends Component {
         this.io.observe(entry);
       });
   }
+
+  changeDataTypeColor(columnName) {
+    this.setState({highlightedDataType: columnName}, () => {
+      setTimeout(() => {
+        this.setState({highlightedDataType: null});
```

Comment:
This might be costly when there are too many rows. Setting state will re-render the entire table. We should try to do this with css animation. Not entirely clear on the solution but we should think about it.

Reply:
this is the only comment not handled yet.

Reply:
fixed. Converted `DataType` to be its own component.